retrieval/discovery/discovery.go

```diff
@@ -41,12 +39,16 @@ func NewKubernetesDiscovery(conf *config.KubernetesSDConfig) (*kubernetes.Discov
 }
 
 // NewMarathon creates a new Marathon based discovery.
-func NewMarathon(conf *config.MarathonSDConfig) *marathon.Discovery {
-	return &marathon.Discovery{
-		Servers:         conf.Servers,
-		RefreshInterval: time.Duration(conf.RefreshInterval),
-		Client:          marathon.FetchApps,
+func NewMarathon(conf *config.MarathonSDConfig) (*marathon.Discovery, error) {
+	md := &marathon.Discovery{
+		Conf:       conf,
+		AppsClient: marathon.FetchApps,
+	}
+	err := md.Initialize()
```

Comment:
I agree, I've got rid of both `Initialize` and `newMarathonHTTPClient`. While `newMarathonHTTPClient` is a pretty small function it will be clearer to keep it inside `NewDiscovery`